packages/hooks_riverpod/test/provider_listener_hook_test.dart

```diff
@@ -0,0 +1,285 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:mockito/mockito.dart';
+import 'utils.dart';
+
+void main() {
+  testWidgets('useProviderListener onChange is invoked', (tester) async {
+    final countProvider = StateProvider((_) => 0);
+    final textProvider = StateProvider((_) => 'initial');
+    final key = GlobalKey();
+    await tester.pumpWidget(ProviderScope(
+      child: HookBuilder(
+        key: key,
+        builder: (context) {
+          final count = useProvider(countProvider).state;
```

Comment:
Why have an UI and listening to providers here?

Reply:
I just wanted to test that onChange is invoked and can interact with other providers as well, but on second thought it doesn't make any sense. I'll remove it